project/apps/api-gateway/src/questions/questions.controller.ts

```diff
@@ -48,10 +58,14 @@ export class QuestionsController {
   @Put(':id')
   async updateQuestion(
     @Param('id') id: string,
-    @Body() updateQuestionDto: UpdateQuestionDto,
+    @Body(new ZodValidationPipe(updateQuestionSchema)) // validation on the body only
+    updateQuestionDto: UpdateQuestionDto,
   ) {
     if (id != updateQuestionDto.id) {
-      throw new Error('ID in URL does not match ID in request body');
+      throw new HttpException(
+        'ID in URL does not match ID in request body',
+        HttpStatus.BAD_REQUEST,
+      );
```

Comment:
would this be different from throwing BadRequestException instead?

Reply:
nope, updated with BadRequestException to match that in auth service